packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -257,9 +260,60 @@ export function getSdkTypeBaseHelper<TKind>(
     __raw: type,
     deprecation: getDeprecationDetails(context.program, type)?.message,
     kind,
+    decorators: getTypeDecorators(context, type),
   };
 }
 
+export function getTypeDecorators(context: TCGCContext, type: Type): Record<string, Array<any>> {
+  const retval: Record<string, Array<any>> = {};
+  if ("decorators" in type) {
+    for (const decorator of type.decorators) {
+      // only process explicitly defined decorators
+      if (decorator.definition) {
+        const decoratorName = `${decorator.definition?.namespace ? getNamespaceFullName(decorator.definition.namespace) + "." : ""}${decorator.definition?.name}`;
+        // white list filtering
+        if (
+          !context.decoratorsWhiteList ||
+          !context.decoratorsWhiteList.some(
+            (x) => x === decoratorName || new RegExp(x).test(decoratorName)
+          )
+        ) {
+          continue;
+        }
+
+        retval[decoratorName] = [];
+        for (const arg of decorator.args) {
+          retval[decoratorName].push(getDecoratorArgValue(arg.jsValue));
+        }
+      }
+    }
+  }
+  return retval;
+}
+
+function getDecoratorArgValue(
+  arg:
+    | Type
+    | Record<string, unknown>
+    | Value
+    | unknown[]
+    | string
+    | number
+    | boolean
+    | Numeric
+    | null
+): any {
+  if (typeof arg === "object" && arg !== null && "kind" in arg) {
+    if (arg.kind === "EnumMember") {
+      return arg.value ?? arg.name;
+    }
+    if (arg.kind === "String" || arg.kind === "Number" || arg.kind === "Boolean") {
+      return arg.value;
+    }
+  }
+  return arg;
```

Comment:
If `arg` is not among the types handling above, we just pass along it to the emitter?
What if `arg` needs some specific complex logic to process it? Does it mean this decorator should not be included in the white list?

Reply:
For those types, we do not have a common way to deal with, so just leave it to emitter.
Yes, I believe those decorators should not belong to the usage of generic decorators.